### ISSUE TITLE
Cannot mark tests as skipped or incomplete in `beforeClass` hooks

### ISSUE THREAD

**Issue by jderusse** [sebastianbergmann/phpunit#4689]
| Q                   | A
| --------------------| ---------------
| PHPUnit version     | 9.5.2
| PHP version         | 8.0
| Installation Method | Composer / PHAR

#### Summary

When calling `self::markTestSkipped` in a `setUpBeforeClass` method, all tests inside the file are flagged `Skipped` without the error "Test skipped because of an error in hook method"

When looking at https://github.com/sebastianbergmann/phpunit/blob/master/src/Framework/TestSuite.php#L438-L448
It looks like we can fix this by throwing a `SkippedTestSuiteError` instead. But this class is `@internal`.

#### Current behavior

```
1) FooTest::testMethod
Test skipped because of an error in hook method
```

#### How to reproduce

```
class FooTest extends TestCase {
    public static function setUpBeforeClass(): void
    {
        self::markTestSkipped('My Custom message');
    }
}
```

#### Expected behavior

```
1) FooTest::testMethod
My Custom message
```

#### Suggestion

Add a public static  `markTestSuiteSkipped` method to the API

**Comment by sebastianbergmann** [comment]
Thank you for your report.

Please provide a **minimal**, **self-contained**, reproducing test case that shows the problem you are reporting.

Without such a minimal, self-contained, reproducing test case I will not be able to investigate this issue.

**Comment by jderusse** [comment]
> Thank you for your report.
> 
> Please provide a **minimal**, **self-contained**, reproducing test case that shows the problem you are reporting.
> 
> Without such a minimal, self-contained, reproducing test case I will not be able to investigate this issue.

Here is a minimal reproducer
```
git clone https://github.com/jderusse/phpunit-4689
cd phpunit-4689
composer install
./vendor/bin/phpunit tests -vvv
```

**Comment by jderusse** [comment]
Please find here a reproducer: https://github.com/jderusse/phpunit-4689

```
git clone https://github.com/jderusse/phpunit-4689
cd phpunit-4689
composer i
./vendor/bin/phpunit tests -v
```

**Comment by nicolas-grekas** [comment]
FYI on Symfony, we merged https://github.com/symfony/symfony/pull/41384 meanwhile.
This might break in the future as `SkippedTestSuiteError` is `@internal`, but that's the best option we have at the moment.

**Comment by sebastianbergmann** [comment]
Superseded by #5164

**Comment by sebastianbergmann** [comment]
I apologize for dropping the ball on this. I split this issue into two new issues, #5164 for PHPUnit 9.6 and #5165 for PHPUnit 10. #5164 is about to be closed, will look into #5165 next.
